numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -1443,6 +1482,10 @@ static PyObject *npyiter_operands_get(NewNpyArrayIterObject *self)
                 "Iterator is invalid");
         return NULL;
     }
+    if (self->open == 0) {
+        PyErr_SetString(PyExc_RuntimeError, "Iterator closed");
```

Comment:
It would be great to add a comment to each of these explaining why it is no longer valid to access the data.

For instance, would there be any harm in exposing a readonly view on the operands here?

Reply:
It's more of a philisophical than practical question. Once the iter is closed, the ``it.operand[0].data`` may (if writeback semantics were involved) fall out of sync with ``op.data``, which could introduce subtle bugs. These bugs might not show up in testing, since it is difficult to predict when writeback semantics will be triggered. I would prefer as much as possible to disallow use of a closed iter, just in case